packages/flutter_tools/bin/xcode_backend.sh

```diff
@@ -129,6 +143,15 @@ BuildApp() {
     StreamOutput " ├─Building Dart code..."
     # Transform ARCHS to comma-separated list of target architectures.
     local archs="${ARCHS// /,}"
+    if [[ $archs =~ .*i386.* || $archs =~ .*x86_64.* ]]; then
+      EchoError "========================================================================"
+      EchoError "ERROR: Flutter does not support running in profile or release mode on"
+      EchoError "the Simulator (this build was: '$build_mode')."
+      EchoError "You can ensure Flutter runs in Debug mode with your host app in release"
```

Comment:
Since there are 2 modes being referred here, make every reference to a mode specific. In your first paragraph, specify 'does not support running the Flutter program in profile or release mode' etc. 

In the second paragraph, specifiy in the .xcconfig of your selected build config.

Reply:
Done